apis/addons/v1alpha1/addons_types.go

```diff
@@ -284,6 +288,9 @@ type AddonStatus struct {
 	// Tracks last reported upgrade policy status.
 	// +optional
 	UpgradePolicy *AddonUpgradePolicyStatus `json:"upgradePolicy,omitempty"`
+	// Tracks the last reported addon status.
+	// +optional
+	ReportedStatus *OCMAddOnStatus `json:"reportedStatus,omitempty"`
```

Comment:
Yeah that sounds like a neat idea 👍

Reply:
@thetechnick 
Do you mean to say that instead of the existing implementation in the PR, the reportedStatus would look like this

```
reportedStatus:
      lastReportedStatusHash: sha256:29jdbsdjee2dbi......
      addonid: foo-addon
      observedGeneration: 42
```
Here the value of `reportedStatus.lastReportedStatus` is formed via `SHA(reported addon status conditions + correlation id)`

If that is NOT what you meant, then you can ignore the following message and correct me.

But in case if you meant this, then imo having a hash might not be seemingly great to have from the POV of the fact that SREs would refer to it `reportedStatus` in the future for potential troubleshooting situations.

Although functionally, the reconcilers can make use of the value of hash to compare and decide whether they need to re-report addon status, yet it would be a hassle for an SRE to make sense out of this hash in situations like troubleshooting an addon. 
Saying this from my personal experience of troubleshooting AddonOperator's upgradePolicy's issue very smoothly by referring to the `.status.upgradePolicy` section.

One could argue that for informational purposes logs should be used but still imo, using hash might hide away any unintended disparities between the hashed reportedStatus and what the logs are reporting.

I agree with your point regarding lower storage footprint on etcd and lower load apiserver but I think that that difference would be negligible in real-life considering that anyway, the number of Addon CRs in any cluster would be way lesser than the number of all the other plethora of resources in that cluster. So, unless `reportedStatus` might increase exponentially under an Addon CR, we won't save much on storage footprint or api-server load by using a hash instead. 

wdyt